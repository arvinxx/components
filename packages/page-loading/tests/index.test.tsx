import React from 'react';
import { render } from '@testing-library/react';

import PageLoading from '@arvinxu/page-loading';

describe('PageLoading', () => {
  it('默认状态', () => {
    const { container } = render(<PageLoading />);
    expect(container).toMatchSnapshot();
  });
  it('不带加载条', () => {
    const { container } = render(<PageLoading progress={false} />);
    expect(container).toMatchSnapshot();
  });
  it('全屏', () => {
    const { container } = render(<PageLoading fullscreen={true} />);
    expect(container).toMatchSnapshot();
  });
  describe('加载态', () => {
    it('加载完毕', () => {
      const { container } = render(
        <PageLoading loading={false}>加载完毕</PageLoading>,
      );
      expect(container).toMatchSnapshot();
    });
    it('加载中', () => {
      const { container } = render(
        <PageLoading loading={true}>加载完毕</PageLoading>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
