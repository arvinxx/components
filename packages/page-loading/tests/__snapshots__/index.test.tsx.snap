// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoading 不带加载条 1`] = `
<div>
  <div
    class="avx-page-loading-container"
    id="avx-page-loading-container"
    style="background: rgb(245, 245, 245);"
  >
    <div
      class="avx-page-loading-loader"
    >
      <div
        class="avx-page-loading-content"
      >
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PageLoading 全屏 1`] = `
<div>
  <div
    class="avx-page-loading-container avx-page-loading-fullscreen nprogress-custom-parent"
    id="avx-page-loading-container"
    style="background: rgb(245, 245, 245);"
  >
    <div
      class="avx-page-loading-loader"
    >
      <div
        class="avx-page-loading-content"
      >
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
      </div>
    </div>
    <div
      class="nprogress"
      id="nprogress"
    >
      <div
        class="bar"
        role="bar"
        style="transition: all 200ms linear; transform: translate3d(-100%,0,0); margin-left: -82%;"
      >
        <div
          class="peg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PageLoading 加载态 加载中 1`] = `
<div>
  <div
    class="avx-page-loading-container nprogress-custom-parent"
    id="avx-page-loading-container"
    style="background: rgb(245, 245, 245);"
  >
    <div
      class="avx-page-loading-loader"
    >
      <div
        class="avx-page-loading-content"
      >
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
      </div>
    </div>
    <div
      class="nprogress"
      id="nprogress"
    >
      <div
        class="bar"
        role="bar"
        style="transition: all 200ms linear; transform: translate3d(-100%,0,0); margin-left: -82%;"
      >
        <div
          class="peg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PageLoading 加载态 加载完毕 1`] = `
<div>
  加载完毕
</div>
`;

exports[`PageLoading 自定义加载图形 1`] = `
<div>
  <div
    class="avx-page-loading-container nprogress-custom-parent"
    id="avx-page-loading-container"
    style="background: rgb(245, 245, 245);"
  >
    <div
      class="avx-page-loading-loader"
    >
      <div>
        loading...
      </div>
    </div>
    <div
      class="nprogress"
      id="nprogress"
    >
      <div
        class="bar"
        role="bar"
        style="transition: all 200ms linear; transform: translate3d(-100%,0,0); margin-left: -82%;"
      >
        <div
          class="peg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`PageLoading 默认状态 1`] = `
<div>
  <div
    class="avx-page-loading-container nprogress-custom-parent"
    id="avx-page-loading-container"
    style="background: rgb(245, 245, 245);"
  >
    <div
      class="avx-page-loading-loader"
    >
      <div
        class="avx-page-loading-content"
      >
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item avx-page-loading-animation-delay"
          style="background: rgb(24, 144, 255);"
        />
        <div
          class="avx-page-loading-item"
          style="background: rgb(24, 144, 255);"
        />
      </div>
    </div>
    <div
      class="nprogress"
      id="nprogress"
    >
      <div
        class="bar"
        role="bar"
        style="transition: all 200ms linear; transform: translate3d(-100%,0,0); margin-left: -82%;"
      >
        <div
          class="peg"
        />
      </div>
    </div>
  </div>
</div>
`;
